src/training/data.py

```diff
@@ -411,13 +519,55 @@ def get_csv_dataset(args, preprocess_fn, is_train, epoch=0, tokenizer=None):
         num_workers=args.workers,
         pin_memory=True,
         sampler=sampler,
+        collate_fn=get_collate_fn(dataset, is_flava),
         drop_last=is_train,
     )
     dataloader.num_samples = num_samples
     dataloader.num_batches = len(dataloader)
 
     return DataInfo(dataloader, sampler)
 
+
+def get_hf_dataset(args, preprocess_fn, is_train, epoch=0, tokenizer=None):
+    dataset_str = args.train_data if is_train else args.val_data
+    identifiers = dataset_str.split(':')
+    assert 1 <= len(identifiers) <= 2, f'Invalid dataset string: {dataset_str}'
+    dataset_name = identifiers[0]
+    subset = identifiers[1] if len(identifiers) == 2 else None
+    split = 'train' if is_train else 'validation'
+    is_flava = args.model.startswith('flava')
```

Comment:
Like mentioned above, this should be taken as preprocess function to be passed to data collator.